### ISSUE TITLE
[a11y]: Aria-hidden incorrectly set on sidebar

### ISSUE THREAD

**Issue by samrobbins85** [carbon-design-system/carbon#9895]
### Package

carbon-components-react

### Browser

Chrome

### Operating System

MacOS

### Package version

7.45.0

### Automated testing tool and ruleset

Axe Devtools

### Description

The sidebar on desktop will have `aria-hidden="true"` even though it's open and so should be accessible

### CodeSandbox example

https://codesandbox.io/s/ibm-sidenav-a11y-jo16k?file=/src/index.js

### Steps to reproduce

1. Open the codesandbox preview in a new window so that it is wide enough to show the sidenav expanded
2. Inspect the `nav` component of the sidebar and see that it has `aria-hidden="true"` but is visible

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
